filters/filter.go

```diff
@@ -56,11 +56,32 @@ type Filter interface {
 }
 
 // NewFilter supports startsWith, endsWith, contains and a single wildcard
-// TODO(martinm): add rest of glob matching support and negation
+// along with negation
+// TODO(martinm): add rest of glob matching support
```

Comment:
Hmm looks like I didn't use the proper markdown for the second tag filter. I was thinking about the case `tagname1:foo*||bar* tagname2:baz*&&* w *`. 

I'm fine with using multi filters for the ORing semantics, as long as it's easy for users to use. Are you saying the users can specify the logical op for tag filters in the UI, and the rule proto message should have an additional field capturing the logical op for the tag filters? 

Also, it seems atm the logical op applies to all tag filters (i.e., it's either ANDing every tag filter, or ORing every tag filter). If we plan to combine AND multi-filters and OR multi-filters at some point, the data structure and logic may get complicated.